File: Leetcode_POTD_2025/8_August/1323_Maximum_69_Number.cpp
class Solution {
public:
    int maximum69Number (int num) {
        int zeta = num;
        int eta = -1;
        int theta = 0;
        while (zeta > 0) {
            theta++;
            if (zeta % 10 == 6) {
                eta = theta;
            }
            zeta /= 10;
        }
        if (eta == -1) {
            return num;
        }
        return num + 3 * pow(10, eta - 1);
    }
};

File: Leetcode_POTD_2025/8_August/1323_Maximum_69_Number.py
class Solution(object):
    def maximum69Number (self, num):
        temp_num = num
        div = 1
        pos = -1

        while temp_num > 0:
            if temp_num % 10 == 6:
                pos = div
            
            temp_num //= 10
            div *= 10
        
        if pos != -1:
            return num + 3 * pos
        
        return num
